chain/network/src/types.rs

```diff
@@ -1198,7 +1196,7 @@ pub enum NetworkViewClientMessages {
     /// Transaction status response
     TxStatusResponse(FinalExecutionOutcomeView),
     /// General query
-    Query { query_id: String, block_id: MaybeBlockId, request: QueryRequest, finality: Finality },
+    Query { query_id: String, block_checkpoint: BlockCheckpoint, request: QueryRequest },
```

Comment:
It did not make sense for `block_id` and `finality` to co-exist. I had to make these changes in this PR as I was extending `changes` API (I should have created a separate PR, but it was hard to decouple the changes).